src/range_proof/dealer.rs

```diff
@@ -295,15 +302,19 @@ impl<'a, 'b> DealerAwaitingProofShares<'a, 'b> {
     /// performing local aggregation,
     /// [`receive_trusted_shares`](DealerAwaitingProofShares::receive_trusted_shares)
     /// saves time by skipping verification of the aggregated proof.
-    pub fn receive_shares(mut self, proof_shares: &[ProofShare]) -> Result<RangeProof, MPCError> {
+    pub fn receive_shares<T: RngCore + CryptoRng>(
```

Comment:
I guess this needs to be a new `receive_shares_with_rng` function, because `receive_shares` is part of the public API and can't change without a major version bump.

Reply:
Ok, I'll rework these so the API doesn't change in std mode.

Reply:
Thanks! Sorry again for missing this on the first pass of review :disappointed: